File: SortWare/InvalidDirectoryException.vb
Public Class InvalidDirectoryException : Inherits System.Exception
    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(ByVal message As String)
        MyBase.New(message)
    End Sub

    Public Sub New(ByVal message As String, ByVal innerException As System.Exception)
        MyBase.New(message, innerException)
    End Sub
End Class

File: SortWare/InvalidDirectoryException.cs
using System;

namespace SortWare
{
    public class InvalidDirectoryException : Exception
    {
        public InvalidDirectoryException() : base()
        {
        }

        public InvalidDirectoryException(string message) : base(message)
        {
        }

        public InvalidDirectoryException(string message, Exception innerException) : base(message, innerException)
        {
        }
    }
}